bokehjs/src/lib/models/callbacks/open_url.ts

```diff
@@ -41,14 +41,28 @@ export class OpenURL extends Callback {
         window.open(url)
     }
 
-    const {selected} = source
+    if (source.data.tap_type == "select")
+    {
+      const {selected} = source
 
-    for (const i of selected.indices)
-      open_url(i)
+      for (const i of selected.indices)
+        open_url(i)
+  
+      for (const i of selected.line_indices)
+        open_url(i)
+  
+      // TODO: multiline_indices: {[key: string]: number[]}
+    } else {
+      const {inspected} = source
 
-    for (const i of selected.line_indices)
-      open_url(i)
-
-    // TODO: multiline_indices: {[key: string]: number[]}
+      for (const i of inspected.indices)
+        open_url(i)
+  
+      for (const i of inspected.line_indices)
+        open_url(i)
+  
+      // TODO: multiline_indices: {[key: string]: number[]}
+    }
+    
```

Comment:
`insepcted` and `selected` are both the same data type. I think this can be simplified and made short, e.g. 
```
const selected = data.behavior == "select" ? source.selected : source.inspected
```
Then the code below can just be all the original code referring to selected without the duplication in separate branches.

Reply:
I am a little confused about how `OpenURL.execute` receives the flag.  In tap_tool.cs, `callback.execute(this.model, data)` passes `data` as an argument to `OpenURL.execute`. However in the `OpenURL.execute`, its second argument is `source` which is of type `ColumnarDataSource`. My question is, would `data.behavior` be able to access the flag, or `source.data.behavior`. Thank you for your time.

Reply:
@dmellogu the second argument is destructuring the passed in `data` value:
```
execute(_cb_obj: unknown, {source}: {source: ColumnarDataSource}): void {
```
I think (untested) that an appropriate new signature would be
```
execute(_cb_obj: unknown, {source, behavior}: {source: ColumnarDataSource, behavior: TapBehavior}): void {

```

Reply:
@bryevdv Thank you for the helpful suggestions. I have updated my branch to reflect these changes.